src/webgpu/web_platform/external_texture/video.spec.ts

```diff
@@ -3,13 +3,150 @@ Tests for external textures from HTMLVideoElement (and other video-type sources?
 
 - videos with various encodings, color spaces, metadata
 
-TODO: get test videos from WebGL CTS
-TODO(kainino0x): (and set up the build to deal with data files)
 TODO: consider whether external_texture and copyToTexture video tests should be in the same file
-TODO: plan
+
+Test Coverage:
+  - Tests that we can import an HTMLVideoElement into a GPUExternalTexture and sample from it for:
+      - All supported video formats {vp8, vp9, ogg, mp4}
+      TODO: Multiplanar scenarios
 `;
 
 import { makeTestGroup } from '../../../common/framework/test_group.js';
 import { GPUTest } from '../../gpu_test.js';
 
+const kHeight = 16;
+const kWidth = 16;
+const kFormat = 'rgba8unorm';
+const kVideoSources: string[] = [
+  'red-green.webmvp8.webm',
+  'red-green.bt601.vp9.webm',
+  'red-green.mp4',
+  'red-green.theora.ogv',
+];
+
 export const g = makeTestGroup(GPUTest);
+
+g.test('importExternalTexture,sample')
+  .desc(
+    `
+Test that importing different video formats into an external texture and sampling from them works correctly.
+`
+  )
+  .params(u =>
+    u //
+      .combine('videoSource', kVideoSources)
+  )
+  .fn(async t => {
+    const videoUrl = '../out/resources/' + t.params.videoSource;
+    const video = document.createElement('video');
+    video.src = videoUrl;
+    if (!video.canPlayType) {
+      throw new Error();
+    }
+
+    video.loop = true;
+    video.muted = true;
+    video.play();
```

Comment:
The state of the art in getting videos to play for tests can usually be found in `startPlayingAndWaitForVideo`:
https://github.com/KhronosGroup/WebGL/blob/main/sdk/tests/js/webgl-test-utils.js
So it might be worth adding a link here to that, in case we ever need to reference it again.

I'm not sure whether we should port in everything that's there right now. It may be the safest bet.

Reply:
I think it makes sense to add in at least a portion of the helper - it seems like it's going to require running each test as a callback, so may as well start in that format to avoid churn later.

Reply:
I think that makes sense. It could be a promise instead of a callback - but I suppose when requestVideoFrameCallback is supported, the callback is probably more likely to give us exactly the right frame (rather than a later one).

Reply:
I've been trying to figure out how use the their function directly - but I'm not very experienced with JS and having issues with async execution. I've updated the PR with startPlayingAndWaitForVideo, however it doesn't function correctly. I find that finalize() is being called before the test callback occurs, which is obviously wrong. Maybe take a look and see if you have a suggestion? If I had a way for startPlayingAndWaitForVideo to wait until maybeCallCallback has finished, we could use this.

Reply:
Resolved.